src/app/CreateRecording/CustomRecordingForm.tsx

```diff
@@ -38,11 +38,12 @@
 import * as React from 'react';
 import { NotificationsContext } from '@app/Notifications/Notifications';
 import { ServiceContext } from '@app/Shared/Services/Services';
-import { ActionGroup, Button, Checkbox, Form, FormGroup, FormSelect, FormSelectOption, FormSelectOptionGroup, Split, SplitItem, Text, TextArea, TextInput, TextVariants, Tooltip, TooltipPosition, ValidatedOptions } from '@patternfly/react-core';
+import { ActionGroup, Button, Checkbox, ExpandableSection, Form, FormGroup, FormSelect, FormSelectOption, FormSelectOptionGroup, Split, SplitItem, Text, TextArea, TextInput, TextVariants, Tooltip, TooltipPosition, ValidatedOptions } from '@patternfly/react-core';
 import { OutlinedQuestionCircleIcon } from '@patternfly/react-icons';
 import { useHistory } from 'react-router-dom';
 import { concatMap } from 'rxjs/operators';
 import { EventTemplate } from './CreateRecording';
+import { RecordingOptions } from '@app/Shared/Services/Api.service';
 
 export interface CustomRecordingFormProps {
   onSubmit: (recordingName: string, events: string, duration?: number) => void;
```

Comment:
I was also wondering if I should modify this interface, I don't know how its used anywhere, but its outdated nonetheless

Reply:
Ah, it should be used to annotate the type of the props expected/received by the CustomRecordingForm component. The "parent" CreateRecording component is an example of this being used properly. Since you're already here, please do go ahead and correct the component definition to specify its prop types with this interface, and update the interface to reflect the actual requirements.

The SnapshotRecordingForm component also has the same problem.

Reply:
I fixed the interface with the proper requirements, Im still a bit confused as to how this props interface works though so I don't know if other things should be added?

Reply:
The interface is used to specify the types that the component receives in its props. This is just used for compile-time type checking.

https://reactjs.org/docs/components-and-props.html

https://github.com/rh-jmc-team/container-jfr-web/blob/8cd7842af44ac78eba8fdf7b18e84c36e1ef98c6/src/app/CreateRecording/CreateRecording.tsx#L49

https://github.com/rh-jmc-team/container-jfr-web/blob/8cd7842af44ac78eba8fdf7b18e84c36e1ef98c6/src/app/CreateRecording/CreateRecording.tsx#L61